lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -839,6 +845,17 @@ void IMConstPropPass::visitOperation(Operation *op, FieldRef changedField) {
     logger.getOStream() << "}\n";
   });
 
+  // If the folding was in-place, mark the results as overdefined and reset
```

Comment:
That's pretty conservative.  This anytime we test an op with a constant on the lhs of a commutative op, we will get overdefined (as the folder will try to move the constant right).

Reply:
Perhaps if it is updated in place, we can keep it and recompute the lattice value?  the commutative folder bases it's reordering on the proposed constants, so it will try in-place updates, but do we care?  We would be assuming it's the only folder that does inplace updates.

Reply:
Yeah I think this is too conservative - no need to mark the results as overdefined here.  Copied and pasted but didn't think too hard about it since everything was still crashing. Also missing a reset above when all the operands have known values.